src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -163,162 +173,598 @@ public long GetNumberOfCountersToProcess(DocumentsOperationContext context, stri
+                                        localCounters.Modifications[counterName] = mergedCv;
+                                        continue;
+
+                                    }
+                                    if (loaclCounterValues != null)
+                                    {
+                                        // delete + blob => resolve conflict
+
+                                        using (var localValuesAsCv = RawBlobToChangeVector(context, data, loaclCounterValues, existingChangeVector))
+                                        {
+                                            var conflictStatus = ChangeVectorUtils.GetConflictStatus(deletedSourceCounter, localValuesAsCv);
+
+                                            switch (conflictStatus)
+                                            {
+                                                case ConflictStatus.Update:
+                                                    //delete is more up do date
+                                                    modified = true;
+                                                    changeType = CounterChangeTypes.Delete;
+                                                    localCounters.Modifications = localCounters.Modifications ?? new DynamicJsonValue(localCounters);
+                                                    localCounters.Modifications[counterName] = deletedSourceCounter;
+                                                    break;
+                                                case ConflictStatus.Conflict:
+                                                // conflict => resolve to raw blob (no change)
+                                                case ConflictStatus.AlreadyMerged:
+                                                    // raw blob is more up to date (no change)
+                                                    continue;
+                                            }
+                                        }                                       
+                                    }
+                                    else
+                                    {
+                                        // put deleted counter
+
+                                        modified = true;
+                                        changeType = CounterChangeTypes.Delete;
+                                        localCounters.Modifications = localCounters.Modifications ?? new DynamicJsonValue(localCounters);
+                                        // todo add local dbId + etag ? 
+                                        localCounters.Modifications[counterName] = deletedSourceCounter;
+                                        _countersCount--;
+                                    }
+                                }
+
+                                if (prop.Value is BlittableJsonReaderObject.RawBlob sourceBlob)
+                                {
```

Comment:
Extract this to a function that operate over delete cv and counter value.
Should't matter which direction

Reply:
done